### DISCUSSION TITLE
HistogramStandardization with masks

### DISCUSSION THREAD

**Discussion by Linardos** [TorchIO-project/torchio#444] [Q&A] [answered]
Is there a way for me to train histogram standardization using a series of masks that define which pixels are to be used per input? Right now the function accepts a single mask path while also accepting a list of mri paths. It seems weird as every image corresponds to its own mask, so I would expect to be inputting a list of paths in the mask case as well. But when I do it I get a TypeError: expected str, bytes or os.PathLike object, not list

```
        self.landmarks = (
            landmarks_path
            if landmarks_path.is_file()
            else tio.HistogramStandardization.train(mri_paths,
                                                    mask_path=gts_paths,
                                                    output_path=landmarks_path)
        )
```
https://torchio.readthedocs.io/transforms/preprocessing.html#torchio.transforms.HistogramStandardization.train

What am I missing here?

**Comment by fepegar** [comment, marked as answer]
Hi, @Linardos. The implementation of this transform is a bit clunky, it's one of the first in the library.

> Is there a way for me to train histogram standardization using a series of masks that define which pixels are to be used per input?

What you're asking for is not supported at the moment, but it shouldn't be difficult to implement and you're welcome to submit a PR for that. Here's the relevant bit of code:

https://github.com/fepegar/torchio/blob/d1c7b8fbef5bc14d232961c1dfc26411b9de3389/torchio/transforms/preprocessing/intensity/histogram_standardization.py#L157-L166

Changes needed:
- Update docstring
- Update typing hints
- Modify the loop to get the mask path for the corresponding image if the input is a list of paths instead of a single path. Maybe a good way to do this is using `enumerate`  to get the index

> But when I do it I get a TypeError: expected str, bytes or os.PathLike object, not list

The docstring says that `mask_path` should be `None`, `str` or `pathlib.Path`, but you passed a `list`, thus the error.
